rules/NetteToSymfony/NodeAnalyzer/NetteControlFactoryInterfaceAnalyzer.php

```diff
@@ -38,12 +34,16 @@ public function isComponentFactoryInterface(Interface_ $interface): bool
                 return false;
             }
 
-            $className = $this->nodeTypeResolver->getFullyQualifiedClassName($returnType);
-            if (is_a($className, 'Nette\Application\UI\Control', true)) {
-                return true;
+            $controlOrForm = new UnionType([
+                new ObjectType('Nette\Application\UI\Control'),
+                new ObjectType('Nette\Application\UI\Form'),
+            ]);
+
+            if ($returnType instanceof ShortenedObjectType) {
+                $returnType = new ObjectType($returnType->getFullyQualifiedName());
```

Comment:
The check work without this wraparound, as in other places

Reply:
without this check, it got error in test like in my previous comment https://github.com/rectorphp/rector/pull/5907#issuecomment-802718573